draft-barnes-mls-protocol.md

```diff
@@ -1435,44 +1434,128 @@ all arrive at the following state:
 
 # Message Protection
 
-[[ OPEN ISSUE: This section has initial considerations about message
-protection.  This issue clearly needs more specific recommendations,
-possibly a protocol specification in this document or a separate
-one. ]]
-
-The primary purpose of this protocol is to enable an authenticated
-group key exchange among participants.  In order to protect messages sent among
-those participants, an application will need to specify how messages are protected.
-
-For every epoch, the root key of the ratcheting tree can be used to
-derive key material for symmetric operations such as encryption/AEAD and MAC;
-AEAD or MAC MUST be used to ensure that the message originated from a member
-of the group.
-
-In addition, asymmetric signatures SHOULD be used to authenticate the sender
-of a message.
+The primary purpose of the handshake protocol is to enable an authenticated
+group key exchange among participants.  In order to protect Application messages sent
+among those participants, an application will need to specify how messages
+are protected. The message protection layer is somewhat independant of the chosen
+group key-establishment mechanism as long as this mechanism provides the
+expected cryptographic security properties for the Message Protection Layer.
+
+All Application messages SHOULD be protected with an Authenticated-Encryption
+with Associated-Data (AEAD) encryption scheme using AEAD keys derived from the
+Application secret (ie. see the key schedule figure).
+Note that "Authenticated" in this context is not mean messages are known to
+be sent by a specific participant but only from a legitimate member of the group.
+To obtain non-repudiability, Handshake messages MUST use asymmetric signatures
+to strongly authenticate the sender of a message, Application messages SHOULD
+use such signatures to provide the same property.
+
+For each change to the Group root secret of the ratcheting tree, participants
+MUST derive new Application secret.
```

Comment:
Isn't this obvious?  When you update the group secret, it replaces the application secret.  I hope you're not envisioning combining the new group secret with the old app secret, because that's more trouble than it's worth.